Effisense/sensor-dashboard
# Create container view, select container dropdown, data flow from `/scan` to `sensors/create`

Closes: #21 

## Checklist


## Changelog

- Add container view. Much of the same functionality as in `/sensors/create`
- Add DropdownMenu
- Add `SelectContainerDropdown` with specific functionality for that dropdown
- Add loading spinner. Use it in `Navigation` when loading state for being signed in and when fetching data in the `SelectContainerDropdown`
- Pass data from `/scan` to `/sensors/create`, and populate the form using the scanned `sensorId` and `collectionId`

## How to test

- Ensure you are signed out
- Visit `/containers/create`. Observe that you are redirect to `/sign-in`
- Sign in
- Visit `/sensors/create`. Observe that you are redirected to `/scan`
- Scan your QR code from Slack. You may also test an incorrect QR code if you want to (this is handled), but that's not the scope of this PR.
- Observe that you are redirected to `/sensors/create`, now with `deviceId` and `collectionId` as query parameters.
- Try to submit the form. Observe that you can't, due to form validation.
- Click around and explore the container dropdown.
- If you do not have any container **in your organization**, click the button to add a new container.
- Observe that you are redirected to `/containers/create`, with the previous query parameters also attached to this URL.
- Add a container as you would like. I recommend adding multiple container, some with short names and long names. That way, you can observe how the dropdown handles the various names.
- After successfully creating a container, observe that you are redirect back to `/sensors/create`, with **both query parameters that you previously had still intact**
- Observe that you can now select a container from the dropdown list. Observe that the name is displayed when setting that container as selected.
- Fill out the rest of the form in `/sensors/create` as you please.
- **To finish this off, you can start Prisma Studio to inspect your new writes to the database. `cd packages/db` and `pnpm studio`. Should start on `localhost:3002`.**
